packages/@o3r/components/builders/metadata-check/helpers/config-metadata-comparison.helper.ts

```diff
@@ -0,0 +1,49 @@
+import type { ComponentConfigOutput } from '@o3r/components';
+import type { MetadataComparator } from '@o3r/extractors';
+
+/**
+ * Interface describing a config migration element
+ */
+export interface MigrationConfigData {
+  /** Library name */
+  libraryName: string;
+  /**
+   * Configuration name
+   */
+  configName?: string;
+  /**
+   * Configuration property name
+   */
+  propertyName?: string;
+}
+
+/**
+ * Returns an array of config metadata from a metadata file.
+ * To be easily parseable, the properties will be split in separate items of the array.
+ * e.g. : [{ library: '@o3r/demo', properties: [{name : 'property1', type: 'string'}, {name : 'property2', type: 'number'}] }]
+ * will become :
+ * [{ library: '@o3r/demo', properties: [{name : 'property1', type: 'string'}] }, { library: '@o3r/demo', properties: [{name : 'property2', type: 'number'}] }]
+ *
+ * @param content Content of a migration metadata files
+ */
+const getConfigurationArray = (content: ComponentConfigOutput[]): ComponentConfigOutput[] => content.flatMap((config) =>
+  config.properties.length > 1
+    ? config.properties.map((prop) => ({...config, properties: [prop]}))
+    : [config]
+);
+
+const getConfigurationPropertyName = (config: ComponentConfigOutput) => `${config.library}#${config.name}` + (config.properties.length ? ` ${config.properties[0].name}` : '');
```

Comment:
missing JSDoc for some methods

Reply:
as discussed the interface where this const is used is documented 
and this const is not exported alone